src/awx_plugins/credentials/plugins.py

```diff
@@ -703,6 +723,14 @@
             'CONTROLLER_PASSWORD': '{{password}}',
             'CONTROLLER_VERIFY_SSL': '{{verify_ssl}}',
             'CONTROLLER_OAUTH_TOKEN': '{{oauth_token}}',
+            'CONTROLLER_REQUEST_TIMEOUT': '{{request_timeout}}',
+            'AAP_HOSTNAME': '{{host}}',
+            'AAP_USERNAME': '{{username}}',
+            'AAP_PASSWORD': '{{password}}',
+            'AAP_VALIDATE_CERTS': '{{verify_ssl}}',
+            'AAP_TOKEN': '{{oauth_token}}',
+            'AAP_REQUEST_TIMEOUT': '{{request_timeout}}',
```

Comment:
These seem fine up to this point.

Reply:
will TOWER_HOST, CONTROLLER_HOST, and AAP_HOSTNAME ever be different?

Reply:
Yes, that is the point. Using CONTROLLER_HOST by itself is using direct API access to the controller host, meaning, you are not going through the proxy. That way of accessing controller is deprecated (exactly the meaning of that word, I'm not the one to ask), so we want to be moving away from that.

However, this change is allowing to use the address of the proxy host. I think @chrismeyersfsu might have expressed a sentiment that this should be a separate type, and there is some validity to that. Ultimately, if we want existing credentials to "just work", we may need some code-level solution to this so that it inspects the response headers or something to determine which it was given, because this is expected to affect the API path that has to be used.

Reply:
@willtome You can point controller_host at the gateway and things work, I tested this all against a gateway 2.5 address like a customer should be doing, The variable itself doesn't care about proxy, endpoint, etc etc. From what I've tested awx.awx against a 2.5 AAP will not work as it does not do the api translation correctly, however ansible.controller collection does. Technically with upstream it could just be Device FQDN, but we have to call it something, and the user relies on the collection to know the difference and know better.